mobile-app/lib/providers/pending_cancellations_provider.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:shared_preferences/shared_preferences.dart';
```

Comment:
Find minimal way to expire this cache after a worst case scenario assumed time, like 5 minutes.